Спринт 17

KateSilencio opened the pull request:
Версия 1

KateSilencio committed:
Спринт 17
Версия 1

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/di/DataModule.kt:
```diff
@@ -0,0 +1,52 @@
+    single<NetworkClient> { RetrofitNetworkClient(ITunesApiCreate = get()) }
+    single<TracksRepository> { TrackRepositoryImpl(networkClient = get()) }
+    //для передачи контекста в sharedPref
+    //single { androidContext() }
```
Закоментированный код лучше сразу удалять. Хранить закоментированный код в проекте - плохая практика, в случае чего этот код можно восстановить с помощью истории гита

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/search/domain/impl/TracksInteractorImpl.kt:
```diff
@@ -5,14 +5,15 @@ import com.example.playlistmaker.search.domain.api.TracksInteractor
     TracksInteractor {
 
-    private val executor = Executors.newCachedThreadPool()
+    //private val executor = Executors.newCachedThreadPool()
```
тут тоже закоментированный лучше почистить

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/search/data/network/RetrofitNetworkClient.kt:
```diff
@@ -4,18 +4,17 @@ import android.util.Log
-class RetrofitNetworkClient: NetworkClient {
-        private val baseUrl = "https://itunes.apple.com"
+class RetrofitNetworkClient(private val ITunesApiCreate: ITunesSearchAPI) : NetworkClient {
+//class RetrofitNetworkClient: NetworkClient {
```
И здесь комментарий

ArturNurtdinov commented on app/src/main/java/com/example/playlistmaker/di/DataModule.kt:
```diff
@@ -0,0 +1,52 @@
+
+// ### Экран настройки ###
+    //контекст уже есть ^^
+    single<SettingsRepository> { SettingsRepositoryImpl(context = get()) }
+    single<ExternalNavigation> { ExternalNavigationImpl(context = get()) }
```
Здесь вместо `context = get()` можно написать сразу `context = androidContext()`, чтобы коин явно сюда подставил application контекст. При использовании `get()` коин будет искать в графе зависимостей соответствующий тип, что явно дольше, чем сразу подставить контекст

ArturNurtdinov reviewed (commented):
1. После нажатия на трек в поиске список треков пропадает и появляется история поиска. Ожидаемое поведение - при клике список треков не показывается, история поиска показывается только при пустом поле ввода:
https://github.com/user-attachments/assets/92b3f735-6142-4484-b47b-9b5c94dd7c0a
2. Если ввести текст в поле ввода и удалить его - подсказка в поле ввода пропадает и не появляется. Ожидаемое поведение - подсказка в поле ввода отображается, если поле ввода пустое:
![image](https://github.com/user-attachments/assets/ee71b790-ad92-46d2-9c5e-860a9705ed77)

KateSilencio committed:
Спринт 17
Версия 2
Изменения:
1. Пыталась добавить в di/dataModule вместо get() AndroidContext(), эмулятор начал подвисать (не знаю почему :/ )

2. Удалены лишние комментарии
3. Добавление подсказки inputEditText.hint в кнопку очистки и в ветку else, если есть фокус и поле ввода пустое
4. В слушатель нажатия трека добавлено searchedTracksView.isVisible = false

KateSilencio committed:
Спринт 17

Добавлена видимость списка поиска в слушатель:
            recyclerView.isVisible = true

Сейчас должно все работать)

KateSilencio committed:
Спринт 18
 Версия 1
